### DISCUSSION TITLE
Way to use PORTB of ATiny87/167 for TX pin

### DISCUSSION THREAD

**Discussion by sergpenshin** [ArminJo/ATtinySerialOut#6] [Q&A] [answered]
Hi @ArminJo.
Thank you for the great repo! 
One question: if this is possible to use PORTB (PB0 to PB7) of ATiny87/167  to define TX pin?
Best regards, Sergey Penshin

**Comment by ArminJo** [comment]
Get the latest version from repository and just use PB0 to PB7 😀

**Reply by sergpenshin** [reply]
Thanks a lot for your answer.
But there is one problem. Maybe because of I'm doing something wrong.

  The `#define TX_PIN PBx` in the sketch does not work as expected. It looks like the` if ((TX_PIN == PA0) || (TX_PIN == PA1) || (TX_PIN == PA2) || (TX_PIN == PA3) || (TX_PIN == PA4) || ( TX_PIN == PA5) || (TX_PIN == PA6) || (TX_PIN == PA7))` is executing, even though `#define TX_PIN PBx` is written in the sketch. TX pin is still on PORTA.   
If only in ATtinySerialOut.hpp is hardcoded (without any condition) like  
```
#define TX_PORT PORTB
#define TX_PORT_ADDR 0x05 // from #define PORTB _SFR_IO8 (0x05)
#define TX_DDR DDRB
```
then it works.

**Comment by ArminJo** [comment, marked as answer]
I am sorry, but you are totally right! I changed the code again, so you have to explicitely define USE_PORTB_FOR_TX_PIN.

**Reply by sergpenshin** [reply]
Thanks a lot. It's great work!
